### ISSUE TITLE
Wrong error "same mangled name" "previous definition is here" is generated for template instantiations (operator overloads?) (maybe "auto" template parameter related)

### ISSUE THREAD

**Issue by m-7761** [llvm/llvm-project#55305]
I'm trying to add Clang to a project's targets with Visual Studio 2022 fresh install, up to /std:c++20 mode. I can just provide this text emitted by these errors because they don't provide any other information. Whatever it is I don't think this error is intended to prevent template instantiations. The "previous definition is here" notes refer back to the exact same line of code.

Logically there can't be a name clash on templates. My hunch is LLVM doesn't fully understand `auto` in template parameters with respect to pointer-to-member constants.

```cpp
		template<int ID, auto PtoM>
		/**TEMPLATE-SPECIALIZATION
		 * ...
		 */
		class Value<ID,bool,PtoM> 
		:
		public InnerValue<ID,bool,PtoM,daeBoolean>
		{
			/*...*/
		};
```

In this case it gets stuck in InnerValue's operator overloads. I think possibly either PtoM isn't properly mangled to be a unique value for different classes/offsets, or it doesn't actually require additional mangling, but the error shouldn't be emitted (assuming some other part of the compiler assures type correctness.) This is the only idea I have.

```console
1>..\dom\include\dae\..\DAEP.h(1572,32): error : definition with same mangled name '??C?$InnerValue@$00_N$0CM@VdaeBoolean@ColladaDOM_3@@@DAEP@ColladaDOM_3@@QBEPBVdaeBoolean@2@XZ' as another definition
1>..\dom\include\dae\..\DAEP.h(1572,32): message : previous definition is here
```

**Comment by davidstone** [comment]
This needs a complete reproducer. Can you add enough code where compiling it shows the error?

**Comment by Endilll** [comment]
Author has never got back to us.
